shtab/__init__.py

```diff
@@ -529,17 +531,22 @@ def recurse(parser, prefix, paths=None):
 
                     # optionals
                     arguments = [
-                        format_optional(opt) for opt in subparser._get_optional_actions()
+                        format_optional(opt, parser) for opt in subparser._get_optional_actions()
                         if opt.help != SUPPRESS]
 
                     # positionals
                     arguments.extend(
-                        format_positional(opt) for opt in subparser._get_positional_actions()
+                        format_positional(opt, parser)
+                        for opt in subparser._get_positional_actions()
                         if not isinstance(opt.choices, dict) if opt.help != SUPPRESS)
 
                     new_pref = f"{prefix}_{wordify(cmd)}"
+                    formatter = subparser._get_formatter()
+                    formatter._width = 1000000
```

Comment:
a bit concerned that this will override a user-specified width. Perhaps save & restore?